src/tui/editor.zig

```diff
@@ -3924,6 +3925,49 @@ pub const Editor = struct {
     }
     pub const to_lower_meta = .{ .description = "Convert selection or word to lower case" };
 
+    fn switch_case_cursel(self: *Self, root_: Buffer.Root, cursel: *CurSel, allocator: Allocator) error{Stop}!Buffer.Root {
+        var root = root_;
+        const saved = cursel.*;
+        const sel = if (cursel.selection) |*sel| sel else ret: {
+            var sel = cursel.enable_selection();
```

Comment:
Should the selection perhaps be disabled again if it was added automatically here?